File: javascript/problem038.js
function toNine(n){
  var a = 1;
  var str = "";
  while(str.length < 9){
    var number = n * a;
    str = str + number;
    a = a + 1;
  }
  if(str.length != 9){
    return false;
  }
  else{
    return str;
  }
}

var pandigitalArray = ['1','2','3','4','5','6','7','8','9'];
function isPandigital(m){
  var returnValue = false;
  if(pandigitalArray.toString() == m.split("").sort().toString()){
    return returnValue = true;
  }
  return returnValue;
}

var maxPandigital = 0;
for(i = 1; i < 10000; i++){
  var nine = toNine(i);
  if(nine != false){
    if(isPandigital(nine)){
      var pandigital = parseInt(nine);
      if(maxPandigital < pandigital){
        maxPandigital = pandigital;
      }
    }
  }
}

var result = maxPandigital;
console.log(result);

File: python/problem038.py
def to_nine(n):
    i = 1
    string = ""
    while len(string) < 9:
        number = n * i
        string += str(number)
        i += 1
    if len(string) != 9:
        return False
    else:
        return string


def is_pandigital(m):
    pandigital_array = ['1', '2', '3', '4', '5', '6', '7', '8', '9']
    temp_array = []
    return_value = False
    for a in m:
        temp_array.append(a)
    temp_array.sort()
    if str(temp_array) == str(pandigital_array):
        return_value = True
    return return_value


index = 1
max_pandigital = 0
while index < 10000:
    nine = to_nine(index)
    if nine is not False:
        if is_pandigital(nine):
            pandigital = int(nine)
            if max_pandigital < pandigital:
                max_pandigital = pandigital
    index += 1

result = max_pandigital
print(result)
